### PR TITLE
Support programatic zoom

### PR DESCRIPTION
**Pull request by masato1230**
## Description
Add two functions that allow zooming to be executed programmatically.
- zoomToOnContentCoordinate
  - zoomTo the position based on content size coordinate. This function can be used to implement the initial zoom functionality.
- zoomToOnLayoutCoordinate
  - zoom to the position based on layout size coordinate. This function can be used to programatic zoom with some user gestures.

## Sample(Capture)
- initial zoom(0~2s)
- zoom with long press(5s~)

https://github.com/usuiat/Zoomable/assets/66544606/ccc8c4a3-48e9-4ce4-9423-30892561864e

Sample code used in video
```
@Composable
fun AsyncImageSample() {
    val zoomState = rememberZoomState()
    val coroutineScope = rememberCoroutineScope()
    AsyncImage(
        model = "https://github.com/usuiat.png",
        contentDescription = "GitHub icon",
        contentScale = ContentScale.Fit,
        onSuccess = { state ->
            zoomState.setContentSize(state.painter.intrinsicSize)
            coroutineScope.launch {
                zoomState.zoomToOnContentCoordinate(zoomTo = Offset(x = 230f, y = 203f))
            }
        },
        modifier = Modifier
            .fillMaxSize()
            .zoomable(zoomState)
            .pointerInput(1) {
                detectTapGestures(
                    onLongPress = {
                        coroutineScope.launch {
                            zoomState.zoomToOnLayoutCoordinate(it)
                        }
                    }
                )
            }
    )
}
```

### CODE REVIEW

**Comment by usuiat** [comment]
Thank you for submitting the PR!
I think it's a good idea to support programatic zoom.
I will review.

**Comment by usuiat** [comment]
The Image should be zoomed around a specified point, but this is not currently the case.

I will explain the case of zoomToOnContentCoordinate using a [sample image](https://github.com/usuiat/Zoomable/blob/main/app/src/main/res/drawable/grid.png).
The image is 512x512 pixel. If I specify Offset(128, 256), the specified point should not move on the screen. But actually it moves.

|Original size|Expected|Actual|
|-|-|-|
|image|image|image|


The same problem exists with zoomToOnLayoutCoordinate.

**Comment by usuiat** [comment]
I have just merged another branch to main branch.
- #60 

In it, there is an internal function called [changeScale](https://github.com/usuiat/Zoomable/blob/a9c7de985f4addabd2412cf59a21ce36733a5c7b/zoomable/src/main/java/net/engawapg/lib/zoomable/ZoomState.kt#L210).
Since changeScale is almost same to zoomToOnLayoutCoordinate, I would like to make this function public and rename it zoomToOnLayoutCoordinate.

And I think zoomToOnContentCoordinate can be achieved by transforming the coordinates and calling zoomToOnLayoutCoordinate.

**Comment by usuiat** [comment]
Sorry for the consecutive posts.
Did you implement it so that the specified point will be centered on the screen?
If so, I was misunderstood.

**Comment by masato1230** [comment]
> Did you implement it so that the specified point will be centered on the screen?

Yes! I have created this Pull Request with the intention of implementing the functionality to center a specified point on the screen.

File: zoomable/src/main/java/net/engawapg/lib/zoomable/ZoomState.kt [hunk 1]
```diff
@@ -233,6 +240,99 @@ class ZoomState(
...
+    /**
+     * Zoom to the position based on content coordinate.
+     * @param zoomTo zoom target position
+     */
+    suspend fun zoomToOnContentCoordinate(
+        zoomTo: Offset,
+        scale: Float = 3f,
+        animationSpec: AnimationSpec<Float> = tween(700),
+    ) = coroutineScope {
+        val fitContentSizeFactor = fitContentSize.width / contentSize.height
```
Comment (usuiat) [inline zoomable/src/main/java/net/engawapg/lib/zoomable/ZoomState.kt:253]: @masato1230 
fitContentSizeFactor is calculated in width / height, but not width / width?

```suggestion
        val fitContentSizeFactor = fitContentSize.width / contentSize.width
```

**Comment by usuiat** [comment]
@masato1230 
I have two requests so that the library users will understand that the specified coordinates move to the center of the screen.
- Change the function name. For example, how about `centerByContentCoordinate(offset, scale, animationSpec)` and `centerByLayoutCoordinate(offset, scale, animationSpec)`?
- Add description to Kdoc of the functions.

**Comment by masato1230** [comment]
@usuiat 
Thank you for your review.
I've made the revisions you requested. Could you please review it again?

**Pull request merged**